nextmv/nextmv/cloud/application.py

```diff
@@ -3174,7 +3174,11 @@ def __input_set_for_scenario(self, scenario: Scenario, scenario_id: str) -> Inpu
         raise ValueError(f"Unknown scenario input type: {scenario.scenario_input.scenario_input_type}")
 
 
-def poll(polling_options: PollingOptions, polling_func: Callable[[], tuple[Any, bool]]) -> Any:
+def poll(  # noqa: C901
+    polling_options: PollingOptions,
+    polling_func: Callable[[], tuple[Any, bool]],
+    __sleep_func: Callable[[float], None] = time.sleep,
```

Comment:
Interesting, I have never seen an argument using the `__` style.

Reply:
This just felt intuitive to me tbf. Happy to workout a different way. I just think that we need to mock away the sleep functionality as the function can't be tested otherwise. Happy to alternatively inject it via some "global" variable if that feels nicer. :relaxed:

Reply:
I think it is fine, was just commenting about the arg name using `_ _ `, no action needed or anything, was just commenting.